src/mixins/selection.js

```diff
@@ -28,7 +28,13 @@ export default {
 
     methods: {
         clearSelection () {
-            this.flatten(this.currentRows).forEach(row => row._meta.selected = false);
+            this.flatten(this.currentRows).forEach(row => {
+                if (row._meta.index !== row._meta.originalIndex) {
+                    row._meta.index = row._meta.originalIndex;
+                }
+                row._meta.selected = false;
+                row._exactMatch = false;
+            });
```

Comment:
This looks like the code that could be shuffling rows. Is it necessary at all?

Reply:
This is necessary to return the original position for the element, to positionwhere it was before filtering.

Reply:
Yeah, I was wrong saying "at all".
Is this the right place for it? We only need to restore the order, when clearing the filter, not a selection. Changing this is optional, only if there a room for improvement.

Reply:
In tthe code there is a `watcher` for filter, this watcher fires every time when filter has changed. In this watcher we call `this.clearSelection()` method, so I think yes, this is the rigth place for it.

Reply:
I mean every time when you are typing something or clearing the filter, eventually we will trigger `clearSelection`